active/0016-emqx-conf.md

```diff
@@ -0,0 +1,184 @@
+# EMQX Configuration Manager
+
+## Changelog
+
+* 2020-10-21: @zhongwencool init draft
+
+## Abstract
+
+Introduces how to  initialize/hot update/reset configurations.
+
+## Motivation
+
+Try to find a suitable way to:
+
+- Hot update configurations via HTTP API.
+- Maintain data consistency for the same configuration, keeping the same configuration in the cluster must be the same, such as `authorization.source`, rules engine's rules.
+- Different nodes of the cluster may set different values for the same configuration item, such as: use different `rate_limit` settings for machines with different hardware limitations.
+
+## Design
+
+### Configuration files
+
+#### emqx.conf
+
+Emqx reads `emqx.conf` forconverting this hocon file into Erlang format at startup, and `emqx.conf` has only 2 lines by default.
```

Comment:
My concern about "only 2 lines"
it is probably going to be a quite large task.
I would recommend NOT to scope it in this EIP.

We can come back and update this one as step 2,
or create another EIP to focus on that change.

Reply:
Create a new EIP should be better.
I update:
...
and `emqx.conf` explicitly include `override.conf` at the end of the file.
....
For now, we need to integrate all the configurations into `emqx.conf`. But it is planned that `emqx.conf` will no longer contain configurations that use default values in the next phase(propose another EIP), the default is only embedded in the code. Also, users can view all configuration items via the HTTP API (described later). This would have 2 benefits: 

- In subsequent version upgrades, adding/removing/updating configurations will not be overwritten by `emqx.conf`. 
- This allows the user to focus only on the configuration that has been modified.